src/framework/components/text/canvas-font.js

```diff
@@ -146,6 +146,18 @@ Object.assign(pc, function () {
         return ctx;
     };
 
+    CanvasFont.prototype._colorToRgbString = function (color, alpha) {
+        var str;
+
+        if (alpha) {
+            str = pc.string.format('rgba({0}, {1}, {2}, {3})', Math.round(255 * color.r), Math.round(255 * color.g), Math.round(255 * color.b), color.a);
```

Comment:
You really shouldn't use pc.string.format. I don't even think it should be public API. Do "rgba(" + r + ", " + g + ", " + b + ")" instead which is 100x faster.

Reply:
And certainly generates less garbage too.

Reply:
https://jsperf.com/playcanvas-format-vs-string-concat/1

Reply:
If you're talking about pc.string.format, yes, it's also ugly (and not readable by JS engineers new to PlayCanvas).